CMakeLists.txt

```diff
@@ -849,11 +842,16 @@ foreach (target ${target_outputs})
 endforeach()
 
 if (BUILD_SHARED)
-  target_link_libraries (libzmq ${CMAKE_THREAD_LIBS_INIT})
-
+  target_link_libraries (libzmq ${OPTIONAL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+  
   if (SODIUM_FOUND)
     target_link_libraries (libzmq ${SODIUM_LIBRARIES})
+	# On Solaris, libsodium depends on libssp
+    if (${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
+      target_link_libraries (libzmq ssp)
```

Comment:
Linking with ssp is only required if libsodium is built as a static library (since it is a private dependency of libsodium). Shouldn't this be handled via the target_link_libraries of the libsodium target automatically?

Reply:
can't remember the details, but on solaris it was a problem even with dynamic linking